src/main/java/com/hotel/mappers/book/request/RoomStaysMapper.java

```diff
@@ -0,0 +1,67 @@
+package com.hotel.mappers.book.request;
+
+import com.hotel.service.book.HotelBookRequest;
+import com.hotel.service.book.PaymentInfo;
+import com.hotel.util.APIConstants;
+import org.opentravel.ota._2003._05.*;
+import org.springframework.stereotype.Component;
+
+import java.math.BigInteger;
+
+@Component
+public class RoomStaysMapper {
+
+    public ArrayOfRoomStaysTypeRoomStay map(HotelBookRequest request) {
+        ArrayOfRoomStaysTypeRoomStay arrayOfRoomStaysTypeRoomStay = new ArrayOfRoomStaysTypeRoomStay();
+        ArrayOfRoomStaysTypeRoomStay.RoomStay roomStay = new ArrayOfRoomStaysTypeRoomStay.RoomStay();
+        ArrayOfRoomTypeType roomType = new ArrayOfRoomTypeType();
+
+        RoomTypeType roomTypeType = new RoomTypeType();
+        roomTypeType.setNumberOfUnits(BigInteger.valueOf(request.getRoomCount()));
+        roomType.getRoomType().add(roomTypeType);
+        roomStay.setRoomTypes(roomType);
+
+        ArrayOfRatePlanType ratePlans = new ArrayOfRatePlanType();
+        RatePlanType ratePlan = new RatePlanType();
+        ratePlan.setRatePlanID(request.getRatePlanId());
+        ratePlans.getRatePlan().add(ratePlan);
+        roomStay.setRatePlans(ratePlans);
+
+        GuestCountType guestCounts = new GuestCountType();
+        GuestCountType.GuestCount guestCount = new GuestCountType.GuestCount();
+        guestCount.setCount(BigInteger.valueOf(request.getGuestCount()));
+        guestCount.setAgeQualifyingCode(APIConstants.AGE_QUALIFIER);
+        guestCounts.getGuestCount().add(guestCount);
```

Comment:
I think guestCounts can be moved to another mapper

Reply:
Do we really need a new class for this 4 line code?